ITensorGaussianMPS/examples/spinless_fermion_pairing_and_instability.jl

```diff
@@ -68,15 +68,15 @@ let
   sympert = 0.5 * (pert + transpose(pert))
   psi = ITensorGaussianMPS.correlation_matrix_to_mps(
     sites,
-    ITensorGaussianMPS.interleave(Complex.(Gamma));
-    eigval_cutof=1e-10,
+    Pairing(ITensorGaussianMPS.interleave(Complex.(Gamma)));
```

Comment:
Can't `Pairing` be deduced automatically by analyzing `length(sites)` and `size(Gamma)`?

Reply:
Yes, the motivation to require it here was to have a common interface for `correlation_matrix_to_mps` and `correlation_matrix_to_gmps` (the latter requires a wrapped correlation matrix). What about adding that check to the version `correlation_matrix_to_mps` for input of type AbstractMatrix, instead of automatically assuming it's number-conserving.

Reply:
Yeah, that makes sense to have a version specialized to `AbstractMatrix` which tries to deduce the wrapper type automatically.